specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,315 @@
+  /// If it is false then the script execution had an unhandled exception which you
+  /// can get via the Exception property.
+  [propget] HRESULT IsSuccess([out, retval] BOOL* value); 
+
+  /// If IsSuccess is true, then this property is the JSON representation of the result of the script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT ResultAsJson([out, retval] LPWSTR* jsonResult);
+
+  /// If IsSuccess is true and the result of script execution is a string, this method provides the value of the string result
+  /// and otherwise returns E_INVALIDARG.
+  HRESULT TryGetResultAsString([out, retval] LPWSTR* stringResult);
+
+  /// If IsSuccess is false, you can use this property to get the unhandled exception thrown by script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT Exception(
+      [out, retval] ICoreWebView2ExecuteScriptException** exception);
+}
+
+/// This is the callback for ExecuteScriptWithResult
+[uuid(CECDD25B-E6E8-4A4E-B890-BBF95932564F), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptWithResultCompletedHandler : IUnknown {
+
+  /// Provides the result of ExecuteScriptWithResult
+  HRESULT Invoke(
+      [in] HRESULT errorCode, 
+      [in] ICoreWebView2ExecuteScriptResult* result);
+}
+
+/// This is the interface for getting string and exception with ExecuteScriptWithResult
+[uuid(67E0B57B-1AC7-4395-9793-5E4EF9C4B7D9), object, pointer_default(unique)]
+interface ICoreWebView2_10 : ICoreWebView2_9 {
+  
+  /// Run JavaScript code from the javaScript parameter in the current
+  /// top-level document rendered in the WebView.
+  /// The result of the execution is returned asynchronously in the CoreWebView2ExecuteScriptResult object
+  /// which has methods and properties to obtain the successful result of script execution as well as any
+  /// unhandled JavaScript exceptions.
+  /// If this method is
+  /// run after the NavigationStarting event during a navigation, the script
+  /// runs in the new document when loading it, around the time
+  /// ContentLoading is run. This operation executes the script even if
+  /// ICoreWebView2Settings::IsScriptEnabled is set to FALSE.
+  HRESULT ExecuteScriptWithResult(
+      [in] LPCWSTR javaScript,
+      [in] ICoreWebView2ExecuteScriptWithResultCompletedHandler* handler);
+}
+```
+
+## .NET and WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2;
+    runtimeclass CoreWebView2ExecuteScriptResult;
+    runtimeclass CoreWebView2ExecuteScriptException;
+
+    runtimeclass CoreWebView2
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_10")]
+        {
+            Windows.Foundation.IAsyncOperation<CoreWebView2ExecuteScriptResult> ExecuteScriptWithResultAsync(String javaScript);
+        }
+    }
+
+    runtimeclass CoreWebView2ExecuteScriptResult
+    {
+        Boolean IsSuccess { get; };
+
+        String ResultAsJson { get; };
```

Comment:
None of the samples cover this.  That's because it's covered by existing ExecuteScriptAsync documentation, right?

Reply:
Correct. ResultAsJson is the same as what you get from ExecuteScriptAsync

Reply:
Please update with old sample applied to this new API for the ref docs.